src/F2_autoreport.py

```diff
@@ -248,6 +248,103 @@ def insert_log_messages(log_dict):
     )
 
 
+def insert_plot(
+        x_data,
+        y_data,
+        plot_type,
+        plot_title=None,
+        x_axis_name=None,
+        y_axis_name=None,
+        id_plot=None,
+        print_only=False,
+):
+    """ Function to create plots using the Plotly library and replace the plots already generated by the
+    matplotlib library, in the web app version of the auto-report.
+    :param print_only: bool, default value is False
+    :param plot_title: str
+    :param y_axis_name: str
+    :param x_axis_name: str
+    :param id_plot: str, id of the graph. needed to work on the plot later
+    :param x_data: list, or pandas series
+    :param y_data: list, or pandas series, or list of lists
+    :param plot_type: type of the plot (line/bar/pie)
+    :return: a plot object
+    """
+    fig = go.Figure()
+    if plot_type == "line":
+        fig.add_trace(
+            go.Scatter(
+                x=x_data,
+                y=y_data,
+                mode="lines",
+                line=dict(color="royalblue", width=2.5),
+            )
+        )
+    styling_dict = dict(
+        showgrid=True,
+        gridwidth=1,
+        zeroline=True,
+        mirror=True,
+        title_font=dict(size=18, family="Courier", color="black"),
+    )
+    fig.update_layout(
+        title=plot_title,
+        xaxis_title=x_axis_name,
+        yaxis_title=y_axis_name,
+        template="simple_white",
+        xaxis=styling_dict,
+        yaxis=styling_dict,
+    )
+    if print_only == False:
+        return html.Div(
```

Comment:
it is good practice to only have one `return` statement: you can do the same with 

```
answer = [....]
if print_only is False:
     answer.append(html.Div(...))
return answer
```

PS: for `False/True` and `None` use `is` or `is not` instead of `==` or `!=`

Reply:
Got it, implemented the changes :)